### PR TITLE
prediction data flow done

### LINKED ISSUE #3: Prediction: mock api integration
- [x] Entity
- [x] Repository
- [x] Usecase
- [x] Bloc
- [x] Model
- [x] RemoteDatasource
- [x] RemoteDatasouceImpl: from `mock/prediction.json`
- [x] RepositoryImpl 
- [x] Configure dependency injection

### PR DESCRIPTION
**Pull request by Tomal9924**
prediction data flow done

### CODE REVIEW

File: lib/core/config/config.dart [hunk 1]
```diff
@@ -9,6 +9,10 @@ import 'package:http/http.dart';
 import 'package:hydrated_bloc/hydrated_bloc.dart';
 import 'package:internet_connection_checker/internet_connection_checker.dart';
 import 'package:path_provider/path_provider.dart';
+import 'package:podcast/features/prediction/data/repositories/repo_impl.dart';
+import 'package:podcast/features/prediction/domain/repositories/repo.dart';
+import 'package:podcast/features/prediction/prediction.dart';
+import 'package:podcast/features/prediction/presentation/bloc/prediction_bloc.dart';
```
Comment (omar-khaium) [inline lib/core/config/config.dart]: use a barrel file to manage the imports

File: lib/core/config/dependencies.dart [hunk 2]
```diff
@@ -86,4 +87,31 @@ Future<void> get _fixtures async {
...
+    ),
+  );
+
+  sl.registerLazySingleton<PredictionRemoteDataSource>(
+    () => PredictionDataSourceImpl(
+      client: sl(),
+    ),
+  );
+
+  sl.registerLazySingleton<PredictionUseCase>(
```
Comment (omar-khaium) [inline lib/core/config/dependencies.dart]: UseCase should be `factory`

Use this snippet for the solution

```dart 
sl.registerFactory(
    () => PredictionUseCase(
      repository: sl(),
    ),
  );
```

File: lib/features/prediction/data/datasources/remote.dart [hunk 3]
```diff
@@ -0,0 +1,5 @@
+import '../models/prediction.dart';
+
+abstract class PredictionRemoteDataSource {
+  Future<PredictionModel> get prediction;
```
Comment (omar-khaium) [inline lib/features/prediction/data/datasources/remote.dart]: Use this snippet as the correct method definition:

```dart
Future<PredictionModel> fetch({
    required String fixtureGuid,
});
```

File: lib/features/prediction/data/datasources/remote_impl.dart [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+  final Client client;
+
+  PredictionDataSourceImpl({
+    required this.client,
+  });
+
+  @override
+  Future<PredictionModel> get prediction async {
+    //! initialize headers
+    // final Map<String, String> headers = {};
```
Comment (omar-khaium) [inline lib/features/prediction/data/datasources/remote_impl.dart:21]: Use this snippet as the correct method definition:

```dart
 // final Map<String, String> headers = {"fixtureGuid": fixtureGuid};
```

File: lib/features/prediction/data/datasources/remote_impl.dart [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
+class PredictionDataSourceImpl extends PredictionRemoteDataSource {
+  final Client client;
+
+  PredictionDataSourceImpl({
+    required this.client,
+  });
+
+  @override
+  Future<PredictionModel> get prediction async {
```
Comment (omar-khaium) [inline lib/features/prediction/data/datasources/remote_impl.dart]: Fix this line based on method override.

File: lib/features/prediction/data/datasources/remote_impl.dart [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+      await rootBundle.loadString('mock/prediction.json'),
+      HttpStatus.ok,
+    );
+
+    final RemoteResponse<Map<String, dynamic>> result = RemoteResponse.parse(
+      response: response,
+    );
+
+    if (result.success) {
+      final data = PredictionModel.parse(result.result!);
```
Comment (omar-khaium) [inline lib/features/prediction/data/datasources/remote_impl.dart]:
```suggestion
      final PredictionModel prediction = PredictionModel.parse(result.result!);
```

File: lib/features/prediction/data/datasources/remote_impl.dart [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+      return data;
```
Comment (omar-khaium) [inline lib/features/prediction/data/datasources/remote_impl.dart]:
```suggestion
      return prediction;
```

File: lib/features/prediction/data/repositories/repo_impl.dart [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+   final NetworkInfo network;
+  final PredictionRemoteDataSource remote;
+
+  const PredictionRepositoryImpl({
+    required this.network,
+    required this.remote,
+  });
+
+  @override
+  Future<Either<Failure, PredictionEntity>> get prediction async {
```
Comment (omar-khaium) [inline lib/features/prediction/data/repositories/repo_impl.dart]: Fix this line based on the me method override

File: lib/features/prediction/domain/repositories/repo.dart [hunk 9]
```diff
@@ -0,0 +1,10 @@
+
+
+import 'package:either_dart/either.dart';
+
+import '../../../../core/shared/shared.dart';
+import '../../prediction.dart';
+
+abstract class PredictionRepository {
+  Future<Either<Failure, PredictionEntity>> get prediction;
```
Comment (omar-khaium) [inline lib/features/prediction/domain/repositories/repo.dart]: Use this snippet as the correct method definition:

```dart
Future<Either<Failure, PredictionEntity>> fetch({
    required String fixtureGuid,
});
```

File: lib/features/prediction/domain/usecases/prediction.dart [hunk 10]
```diff
@@ -0,0 +1,14 @@
...
+
+import '../../../../core/shared/shared.dart';
+import '../../prediction.dart';
+
+class PredictionUseCase {
+  final PredictionRepository repository;
+  PredictionUseCase({
+    required this.repository,
+  });
+   Future<Either<Failure,PredictionEntity>> call() async {
```
Comment (omar-khaium) [inline lib/features/prediction/domain/usecases/prediction.dart]: Use this snippet as the correct method definition:

```dart
Future<Either<Failure, PredictionEntity>> call({
    required String fixtureGuid,
});
```

File: lib/features/prediction/domain/usecases/prediction.dart [hunk 11]
```diff
@@ -0,0 +1,14 @@
...
+    return await repository.prediction;
```
Comment (omar-khaium) [inline lib/features/prediction/domain/usecases/prediction.dart]:
```suggestion
    return await repository.fetch(fixtureGuid: fixtureGuid);
```

File: lib/features/prediction/presentation/bloc/prediction_bloc.dart [hunk 12]
```diff
@@ -1,13 +1,22 @@
...
 
 class PredictionBloc extends Bloc<PredictionEvent, PredictionState> {
-  PredictionBloc() : super(PredictionInitial()) {
-    on<PredictionEvent>((event, emit) {
-      // TODO: implement event handler
+  final PredictionUseCase useCase;
+  PredictionBloc({required this.useCase}) : super(const PredictionInitial()) {
+    on<FetchPrediction>((event, emit) async {
+      emit(const PredictionLoading());
+      final result = await useCase();
```
Comment (omar-khaium) [inline lib/features/prediction/presentation/bloc/prediction_bloc.dart]:
```suggestion
      final result = await useCase(fixtureGuid: fixtureGuid);
```

File: lib/features/prediction/presentation/bloc/prediction_event.dart
Comment (omar-khaium) [inline lib/features/prediction/presentation/bloc/prediction_event.dart:1]: Use this snippet as the correct method definition:

```dart
class FetchPrediction extends PredictionEvent {
    required String fixtureGuid,

    const FetchPrediction({
        required this.fixtureGuid,
    });
});
```

**Review by omar-khaium — CHANGES_REQUESTED** [review]
Let's fix the issues. most of the things looks good.

File: lib/core/config/dependencies.dart (hunk 2 above)
Comment (Tomal9924) [reply lib/core/config/dependencies.dart]: done

File: lib/core/config/config.dart (hunk 1 above)
Comment (Tomal9924) [reply lib/core/config/config.dart]: Done

File: lib/features/prediction/domain/repositories/repo.dart (hunk 9 above)
Comment (Tomal9924) [reply lib/features/prediction/domain/repositories/repo.dart]: Done

File: lib/features/prediction/data/repositories/repo_impl.dart (hunk 8 above)
Comment (Tomal9924) [reply lib/features/prediction/data/repositories/repo_impl.dart]: Done

File: lib/features/prediction/data/datasources/remote_impl.dart (hunk 5 above)
Comment (Tomal9924) [reply lib/features/prediction/data/datasources/remote_impl.dart]: Done

File: lib/features/prediction/data/datasources/remote_impl.dart (hunk 7 above)
Comment (Tomal9924) [reply lib/features/prediction/data/datasources/remote_impl.dart]: Done

File: lib/features/prediction/presentation/bloc/prediction_event.dart
Comment (Tomal9924) [reply lib/features/prediction/presentation/bloc/prediction_event.dart:1]: Done

File: lib/features/prediction/data/repositories/repo_impl.dart [hunk 13]
```diff
@@ -14,10 +13,12 @@ class PredictionRepositoryImpl implements PredictionRepository {
   });
 
   @override
-  Future<Either<Failure, PredictionEntity>> get prediction async {
-if (await network.online) {
+  Future<Either<Failure, PredictionEntity>> fetchPrediction({
```
Comment (omar-khaium) [inline lib/features/prediction/data/repositories/repo_impl.dart]:
```suggestion
  Future<Either<Failure, PredictionEntity>> fetch({
```

File: lib/features/prediction/domain/repositories/repo.dart [hunk 14]
```diff
@@ -6,5 +6,7 @@ import '../../../../core/shared/shared.dart';
 import '../../prediction.dart';
 
 abstract class PredictionRepository {
-  Future<Either<Failure, PredictionEntity>> get prediction;
+  Future<Either<Failure, PredictionEntity>> fetchPrediction({
```
Comment (omar-khaium) [inline lib/features/prediction/domain/repositories/repo.dart]:
```suggestion
  Future<Either<Failure, PredictionEntity>> fetch({
```

**Review by omar-khaium — CHANGES_REQUESTED** [review]
Two more changes to go.

**Review by omar-khaium — APPROVED** [review]
LGTM

**Pull request merged**